### DISCUSSION TITLE
Do validation with file mimetype?

### DISCUSSION THREAD

**Discussion by SupertigerDev** [kartikk221/hyper-express#301] [Q&A]
When I do this mimetype validation check, there are different things happening on windows and linux.

On Windows: the request responds only sometimes, and other times it times out.
On Linux: It doesnt seem to be working at all, it always times out.

The console log "test" is ran on both systems. just the request is not ending for some reason.
```ts
const server = new Server();

server.post("/", async (req, res) => {
  await req.multipart((field) => {
    if (!field.file) {
      res.send("no file");
      return;
    }
    // do mime checks
    const isValidMime = false;
    if (!isValidMime) {
      console.log("test")
      res.send("invalid file");
      return;
    }
    // handle the file
  });
});

server.listen(3000, () => {
  console.log("Server started on port 3000");
});

```

![image](https://github.com/user-attachments/assets/39262c15-2bd1-41c3-a638-1ebab31c1e44)

**Comment by SupertigerDev** [comment]
Interestingly if i do it like this, it works:

```ts
if (!isValidMime) {
  field.file.stream.on("readable", () => {
    res.status(400).json({
      error: "Invalid image mime type",
    });
  })
  return;
}
```
If this is a good way, please let me know

**Reply by kartikk221** [reply]
Hi, yes. The reason why it hangs is likely because the stream is not being consumed. So when it comes to HTTP requests you have to accept all of the incoming data (You can ignore it but it will still need to be received) before gracefully sending a response.

I think we can possibly treat this as a bug since `hyper-express` should automatically begin flushing the stream If the file is not handled but yeah your solution seems to backup the behavior above.
